src/shared/sam/cli/defaultValidatingSamCliProcessInvoker.ts

```diff
@@ -1,61 +0,0 @@
-/*!
- * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- * SPDX-License-Identifier: Apache-2.0
- */
-
-import { ChildProcessResult } from '../../utilities/childProcess'
-import {
-    DefaultSamCliProcessInvoker,
-    resolveSamCliProcessInvokerContext,
-    SamCliProcessInvokerContext,
-} from './samCliInvoker'
-import { SamCliProcessInvokeOptions, SamCliProcessInvoker } from './samCliInvokerUtils'
-import { throwAndNotifyIfInvalid } from './samCliValidationUtils'
-import {
-    DefaultSamCliValidator,
-    DefaultSamCliValidatorContext,
-    SamCliValidator,
-    SamCliValidatorResult,
-} from './samCliValidator'
-
-/**
- * Validates the SAM CLI version before making calls to the SAM CLI.
- */
-export class DefaultValidatingSamCliProcessInvoker implements SamCliProcessInvoker {
```

Comment:
This class is only used by `src/shared/sam/localLambdaRunner.ts`, and it has a lot of problems. In particular it throws an exception based on user configurable (presence or absence of `sam`), which then causes the extension to fail.

Where we want to show warnings about SAM CLI issues, we should call the sam-detect stuff explicitly.